oremda/clients/base/image.py

```diff
@@ -1,16 +1,26 @@
 from abc import ABC, abstractmethod
 
+from pydantic.error_wrappers import ValidationError
+
 from oremda.clients.utils import flat_get_item, flat_to_nested
+from oremda.typing import OperatorLabels
 
 
 class ImageBase(ABC):
 
     @property
     @abstractmethod
-    def labels(self):
+    def raw_labels(self):
         pass
 
     @property
-    def oremda_labels(self):
-        oremda_labels = flat_get_item(self.labels, 'oremda')
-        return flat_to_nested(oremda_labels)
+    def labels(self) -> OperatorLabels:
+        labels = flat_get_item(self.raw_labels, 'oremda')
+        return OperatorLabels(**flat_to_nested(labels))
+
+    def validate_labels(self):
+        try:
+            self.labels
+        except ValidationError as e:
+            msg = f'Required labels are missing under the oremda prefix:\n{e}'
+            raise Exception(msg) from None
```

Comment:
This `validate_labels()` function calls `self.labels`, and if an exception occurs in pydantic, it simply tries to make the exception more easily understandable.